### ISSUE TITLE
Adding an attachment with :mime_type overrides its :encoding

### ISSUE THREAD

**Issue by fabiob** [mikel/mail#123]
Hello,

Today I stumbled into an issue with attachment support on Mail. I was trying to send some signed XML files using base64 encoding. So I added the following hash onto the attachments array:

```
attachments[my_short_xml_filename] = {
  :mime_type => 'text/xml',
  :data => File.read(my_xml_filename)
}
```

Even though base64 is the default encoding, looks like Mail was changing it to 7bit. So I tried to force the encoding:

```
attachments[my_short_xml_filename] = {
  :mime_type => 'text/xml',
  :encoding => 'base64',
  :data => Base64.encode64( File.read(my_xml_filename) )
}
```

Still no luck, no matter what I did, the encoding keep changing to 7bit. So I took a look at git and found the offending code here: http://github.com/mikel/mail/blob/master/lib/mail/attachments_list.rb#L62

In short, the :mime_type setting overrides any encoding previously set. I was able to make my code work when I removed the :mime_type entry:

```
attachments[my_short_xml_filename] = {
  # :mime_type => 'text/xml',
  :encoding => 'base64',
  :data => Base64.encode64( File.read(my_xml_filename) )
}
```

Hope that helps you guys, I'm kinda in a hurry and can't make a fork with the fix.
